src/com/thunder_cut/server/Process.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Processing.java
+ * Author : Arakene
+ * Created Date : 2020-02-14
+ */
+package com.thunder_cut.server;
+
+import com.thunder_cut.server.data.Commands;
+import com.thunder_cut.server.data.DataType;
+import com.thunder_cut.server.data.ReceivedData;
+import com.thunder_cut.server.data.SendingData;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.*;
+import java.util.function.BiConsumer;
+import java.util.function.Consumer;
+
+/**
+ * After Receiving data, work by data type
+ */
+public class Process {
+
+    private Map<DataType, BiConsumer<ReceivedData, Map<ClientInfo, List<ClientInfo>>>> processMap;
+    private Consumer<ClientInfo> disconnect;
+
+    public Process(Consumer<ClientInfo> disconnect) {
+        processMap = new EnumMap<DataType, BiConsumer<ReceivedData, Map<ClientInfo, List<ClientInfo>>>>(DataType.class);
+        processMap.put(DataType.CMD, this::command);
+        processMap.put(DataType.MSG, this::message);
+        processMap.put(DataType.IMG, this::image);
+        this.disconnect = disconnect;
+    }
+
+    /**
+     * This work if type is command
+     * Split command and work with given command in command process
+     *
+     * @param data      this have data, type, src
+     * @param clientMap client list
+     */
+    private void command(ReceivedData data, Map<ClientInfo, List<ClientInfo>> clientMap) {
+        ByteBuffer buffer = data.getBuffer();
+        buffer.flip();
+        String command = new String(buffer.array());
```

Comment:
You should specify the encoding.

e.g. `new String(buffer.array(), StandardCharset.UTF_8);`

Reply:
I applied encoding